subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/ProjectSchemaProvider.kt

```diff
@@ -46,18 +46,18 @@ typealias TypedProjectSchema = ProjectSchema<SchemaType>
 
 
 data class ProjectSchema<out T>(
-    val extensions: List<ProjectSchemaEntry<T>>,
-    val conventions: List<ProjectSchemaEntry<T>>,
-    val tasks: List<ProjectSchemaEntry<T>>,
-    val containerElements: List<ProjectSchemaEntry<T>>,
+    val extensions: Set<ProjectSchemaEntry<T>>,
+    val conventions: Set<ProjectSchemaEntry<T>>,
+    val tasks: Set<ProjectSchemaEntry<T>>,
+    val containerElements: Set<ProjectSchemaEntry<T>>,
```

Comment:
:x: Let's not change `ProjectSchema` contract but only the algorithm to collect it